### DISCUSSION TITLE
Retry for specific HTTP status?

### DISCUSSION THREAD

**Discussion by gsakkis** [hynek/stamina#69] [Ideas]
I want to retry http request errors with status 500+ (but not 400-500), is this possible? The `on` parameter apparently accepts only exception type(s).

**Comment by hynek** [comment]
Don’t most HTTP libraries have a specific exception sub type for server errors? Otherwise this is a sub type of #63, I think?

**Reply by gsakkis** [reply]
`httpx` has just `HTTPStatusError`:
```py
class HTTPStatusError(HTTPError):
    """
    The response had an error HTTP status of 4xx or 5xx.
```

Yes #63 would cover it. I could work on a PR if that's ok.

**Reply by hynek** [reply]
Sure, I've _just_ updated https://github.com/hynek/stamina/blob/main/.github/CONTRIBUTING.md

**Reply by hynek** [reply]
Thanks for the PR!
